components/sup/src/supervisor.rs

```diff
@@ -151,34 +151,14 @@ impl Supervisor {
 
     /// Send a SIGTERM to a process, wait 8 seconds, then send SIGKILL
     pub fn stop(&mut self) -> Result<()> {
-        let wait = match self.child {
-            Some(ref child) => {
-                let ref pid = child.id();
-                outputln!(preamble & self.preamble, "Stopping");
-                try!(signals::send_signal(*pid, signals::Signal::SIGTERM as u32));
-                true
+        match self.child {
+            Some(ref mut child) => {
+                outputln!(preamble & self.preamble, "Stopping...");
+                try!(child.kill());
             }
-            None => false,
+            None => {},
         };
-        if wait {
-            let stop_time = SteadyTime::now() + Duration::seconds(8);
-            loop {
-                try!(self.check_process());
-                if SteadyTime::now() > stop_time {
-                    outputln!(preamble & self.preamble,
-                              "Process failed to stop with SIGTERM; sending SIGKILL");
```

Comment:
We might want this output in the newly refactored code to update the log with what signal was the final one sent to the process